UI redesign: Change background color.

evykassirer opened the pull request:
Fixes: #21750

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
in comments below

**Self-review checklist**

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [ ] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.

terpimost commented:
I'm ok with any intermediate result here.

alya commented:
Thanks! From playing around for a bit, I see one bug, and a few things to discuss.

alya commented:
The bug is the bit of white background you can see on the right in narrow windows. That stripe also seems to affect the color of the "users" button on the right.

![Screen Shot 2022-12-02 at 10 17 42 AM](https://user-images.githubusercontent.com/2090066/205360238-8e6983ce-b0e9-4b85-8a0c-11d3d91dabcb.png)

alya commented:
The date lines within message blocks (which are the ones we're keeping in the date line redesign) look off to me:

![Screen Shot 2022-12-02 at 10 16 05 AM](https://user-images.githubusercontent.com/2090066/205360834-8981360c-9488-4af2-b3aa-d02b637f589a.png)

@terpimost were those intended to have a grey background, or a white one?

We should also avoid having a different thickness of colored line on the left.

alya commented:
I think @evykassirer asked about this elsewhere, but @terpimost , do you have a recommendation for how to manage the compose box? Having an outline that looks so faint isn't working, I think. Should we make it darker?

Also, looking at screenshots from the redesign proposal, the compose box there has a different grey background color than the sidebars. Can we try switching to that?

![Screen Shot 2022-12-02 at 10 17 17 AM](https://user-images.githubusercontent.com/2090066/205361028-10c81876-1300-437e-959c-ec7d47b336e1.png)

alya commented:
Should we try giving the users list a grey background in narrow windows? That would probably feel more consistent.
![Screen Shot 2022-12-02 at 10 18 52 AM](https://user-images.githubusercontent.com/2090066/205361619-c6bec44d-c24a-4d80-b1e3-560e4f7ee233.png)

terpimost commented:
> @terpimost were those intended to have a grey background, or a white one?

The background of the date separator should be the main background color for the message feed, which is a base background color for not highlighted message.

>do you have a recommendation for how to manage the compose box? Having an outline that looks so faint isn't working, I think

the suggestion is to make background of the compose box a bit darker, something like this: 
<img width="944" alt="image" src="https://user-images.githubusercontent.com/1903309/205933448-f45bbc0c-ee05-436b-9859-fbfd966340b6.png">

>Should we try giving the users list a grey background in narrow windows? That would probably feel more consistent.

My expectation was that it is always grey 
<img width="207" alt="image" src="https://user-images.githubusercontent.com/1903309/205933622-d9d47ef2-cd11-4836-8ee9-66dcc7c1983a.png">

alya commented:
Thanks for the update!

The preview mode color needs to be fixed in dark theme:

<img width="957" alt="Screen Shot 2022-12-06 at 2 49 47 PM" src="https://user-images.githubusercontent.com/2090066/206041287-3d5d07a4-d62f-43ac-bdea-b6b0170a998b.png">

alya commented:
@terpimost How should the compose box look in dark theme? Should it have a different color background?

<img width="1142" alt="Screen Shot 2022-12-06 at 2 51 05 PM" src="https://user-images.githubusercontent.com/2090066/206041447-283d7ab2-8b71-426c-8b99-34b12770d9c5.png">

alya commented:
The light theme compose background looks too dark to me right now, but I'm happy to tune that later.

evykassirer commented:
Dark mode currently seems to have the same color for compose box and the background, but happy to change that if desired.

Agreed that light theme compose background looks dark -- maybe Vlad has a suggestion for another color?

I've addressed the rest of the comments.

alya commented:
@timabbott There are a some compose box color details still being hashed out (see comment just above), but I think it would be good for you to take a look at the implementation at this point.

evykassirer commented:
The commit separation isn't anything intelligible right now. 

I'm actually curious what would be most helpful to Tim in terms of commits -- would one big commit with most of the style changes work (with possibly some prep commits)? Since it's not really the kind of thing we'd want to half-merge.

evykassirer commented on static/styles/compose.css:
```diff
@@ -52,11 +58,11 @@
-    border-top: 1px solid hsla(0, 0%, 0%, 0.07);
+    border: 1px solid hsla(0, 0%, 0%, 0.07);
+    border-bottom: none;
+    background-color: hsl(0, 0%, 85%);
```
Likely will be changed to be less dark, placeholder for now.

evykassirer commented on static/styles/compose.css:
```diff
@@ -89,6 +95,10 @@
     .stream-selection-header-colorblock {
+        box-shadow: none;
+        border: 1px solid hsl(0, 0%, 80%);
+        border-right: none;
```
Changed from box-shadow to border to match the stream name, since the discrepancy was more obvious after the background change

evykassirer commented on static/styles/compose.css:
```diff
@@ -229,8 +239,6 @@
     z-index: 4;
     width: 100%;
-
-    background-color: hsl(0, 0%, 100%);
```
Any selector that has its old background color removed is now inheriting the new background color.

evykassirer commented on static/styles/compose.css:
```diff
@@ -613,7 +622,7 @@ input.recipient_box {
 
     .divider {
-        color: hsl(0, 0%, 88%);
+        color: hsl(0, 0%, 75%);
```
Darker because the composebox itself is now darker. Could also change as we finalize composebox colors.

evykassirer commented on static/styles/compose.css:
```diff
@@ -685,7 +694,7 @@ a.compose_control_button.hide {
     border: 1px solid hsl(0, 0%, 67%);
     border-radius: 4px;
-    background-color: hsla(0, 0%, 0%, 0.02);
+    background-color: hsl(0, 0%, 98%);
```
Switched from transparent overlay to the exact color it was before, since the background behind it changed.

evykassirer commented on static/styles/zulip.css:
```diff
@@ -1373,7 +1390,7 @@ td.pointer {
 
     .message-header-contents {
-        background-color: hsla(192, 19%, 75%, 0.2);
+        background-color: hsl(104.2, 100%, 92.5%);
```
Temporary color -- the actual recipient header is being worked on in a separate PR.

evykassirer commented on static/styles/dark_theme.css:
```diff
@@ -767,8 +798,8 @@
-    #subscription_overlay .settings-radio-input-parent,
-    #stream_privacy_modal .settings-radio-input-parent,
+    #subscription_overlay .radio-input-parent,
+    #stream_privacy_modal .radio-input-parent,
```
oh this seems like a bug, I should add that back

evykassirer commented on static/styles/zulip.css:
```diff
@@ -533,8 +547,10 @@ p.n-margin {
+        height: 100%;
         position: fixed;
-        margin-top: $sidebar_top;
+        padding-top: $sidebar_top;
```
TODO: explanation for why I made the height and margin->padding change

evykassirer commented on static/styles/zulip.css:
```diff
@@ -905,6 +921,7 @@ li.actual-dropdown-menu i {
        in Firefox only, for some reason;
        otherwise it gets a scrollbar */
     overflow: visible;
+    background-color: hsla(0, 0%, 100%, 1);
```
this can be just `hsl`

evykassirer commented on static/styles/zulip.css:
```diff
@@ -1386,7 +1403,7 @@ td.pointer {
     .date_row {
         background-color: hsla(192, 19%, 75%, 0.2);
-        box-shadow: inset 2px 0 0 0 hsl(0, 0%, 27%), -1px 0 0 0 hsl(0, 0%, 27%);
+        box-shadow: inset 0 0 0 0 hsl(0, 0%, 27%), -1px 0 0 0 hsl(0, 0%, 27%);
```
same as above: 

> I should actually probably find a way to not make the stream bar thinner at all, or at least put that in a separate commit.

evykassirer commented on static/styles/zulip.css:
```diff
@@ -1395,7 +1412,7 @@ td.pointer {
     justify-content: space-between;
     border-right: 1px solid hsl(0, 0%, 88%);
-    background-color: hsl(0, 0%, 88%);
+    background-color: hsl(104.2, 100%, 92.5%);
```
Temporary color -- the actual recipient header is being worked on in a separate PR.

evykassirer replied:
Temporary color -- the actual recipient header is being worked on in a separate PR.

evykassirer commented on static/styles/zulip.css:
```diff
@@ -1805,11 +1824,6 @@ div.focused_table {
-.top-navbar-border {
-    border-right: 1px solid hsl(0, 0%, 88%);
-    border-left: 1px solid hsl(0, 0%, 88%);
-}
```
TODO: why is this removed?

evykassirer commented on static/templates/navbar.hbs:
```diff
@@ -1,73 +1,75 @@
-                </div>
-                <div class="top-navbar-border top-navbar-container">
-                    <div id="message_view_header" class="notdisplayed">
+    <div class="header-wrapper">
```
This change only adds this line and a closing tag below (can see if you view the diff hiding whitespace)

timabbott replied:
Can you talk about why we need this `div`? I ask mainly because we've been accused in the past of having endless unnecessary layers of divs in our templates, so I'd like to understand why we can't just do this by applying classes to `header-main` or the `header` element around it.

(I'm also not sure about the naming scheme of `header` contains `header-wrapper` contains `header-main`, but that's probably secondary).

If we indeed need such an outer div, we should consider doing a prep commit to add it (with no visual changes) just to make the diff easier to read.

evykassirer replied:
Ah I inherited that change in this PR and hadn't thought about it critically. Great point, and yes it's not needed so I have removed it :)

evykassirer commented on static/styles/right_sidebar.css:
```diff
@@ -199,7 +199,6 @@ $user_status_emoji_width: 24px;
     right: 0;
     text-align: center;
     vertical-align: middle;
-    border-left: 1px solid hsl(0, 0%, 88%);
```
We're now avoiding lines between items in the header bar. This removes some of that border.

evykassirer commented on static/styles/zulip.css:
```diff
@@ -1373,7 +1390,7 @@ td.pointer {
 
     .message-header-contents {
-        background-color: hsla(192, 19%, 75%, 0.2);
+        background-color: hsl(104.2, 100%, 92.5%);
```
Temporary color -- the actual recipient header is being worked on in a separate PR.

evykassirer replied:
@alya we actually probably shouldn't merge this PR until we have a color for the recipient header, how's the progress on that going?

evykassirer replied:
or maybe this is fine?

<img width="651" alt="image" src="https://user-images.githubusercontent.com/5634097/217739753-ed802528-67e2-418a-9ca4-d2f460293cac.png">

alya replied:
@amanagr Would know best about the recipient header colors PR status. But in any case the next step is to test on CZO, which I think we can do without getting the header bars updated.

evykassirer commented:
PR is now cleaned up and ready for review (once CI passes)

terpimost commented:
@evykassirer here are my suggestions:
dark mode:
<img width="1554" alt="image" src="https://user-images.githubusercontent.com/1903309/206872283-2846b31a-626f-4bee-9a03-5396b257b93b.png">

light mode:
<img width="1556" alt="image" src="https://user-images.githubusercontent.com/1903309/206872420-e0c196b2-47c8-4e92-a9e5-e72a35d04b74.png">

terpimost commented:
@evykassirer I'm sorry for the previous confusion, here is the suggestion for the updated background:
![image](https://user-images.githubusercontent.com/1903309/206884280-839b0797-dffd-4378-b2dd-ca1990c223fe.png)
![image](https://user-images.githubusercontent.com/1903309/206884284-fa61eb7c-f86e-4e96-a4d8-bb01d91ee6e7.png)

evykassirer commented:
Thanks Vlad, made those changes! I believe this is ready for Tim to look at now :)

evykassirer commented:
@timabbott This is ready for you now! Let me know if I can do anything to make it easier to review.

timabbott commented on static/styles/compose.css:
```diff
@@ -1,3 +1,9 @@
+Common background color
+TODO: import from a common place
+*/
+$background_grey: hsl(0, 0%, 94%);
```
Agreed. This may be a good discussion for #frontend -- we should possibly initialize a near-empty file `app_colors.css` that's imported everywhere as a prep PR, so that things like this can just add lines to that.

alya commented:
Removing the "integration review" label, as "chat.zulip.org review" is the equivalent for PR we know we'll want to test on CZO prior to merging.

evykassirer commented:
Conflicts rebased, next step is Tim testing on CZO.

timabbott commented:
This version seems to have a thin white border at the top of recipient bars:

![image](https://user-images.githubusercontent.com/2746074/225458646-599b906e-9fc6-483e-b265-9f7da1926156.png)

I also see a kinda weird background for the top navbar area only that seems possibly not intentional?
![image](https://user-images.githubusercontent.com/2746074/225458786-32dc3089-79b1-46c8-9076-dcd347266d16.png)

I guess my other question is going to be whether this rebases cleanly with #23872 -- I notice a few comments about temporary changes that probably could be skipped if we just stack this on top of that PR here.

evykassirer commented:
Okay, I replaced the [commit based off of main](71f78225e4c9e1390a08a0157eaee63b0f6fda5c) with a commit based off of https://github.com/zulip/zulip/pull/23782!

evykassirer commented:
Resolved conflicts again!

@amanagr there's an issue with the blue box now because the messages overlap each other

<img width="878" alt="image" src="https://user-images.githubusercontent.com/5634097/226764206-0be2ea1b-5bd5-408b-a3cf-4a483b9d501a.png">


What do you think would be a good solution here?

amanagr commented:
I pushed a fix, see https://chat.zulip.org/#narrow/stream/101-design/topic/UI.20redesign.3A.20blue.20box/near/1530162

evykassirer commented:
Sweet, pulled that in:

<img width="918" alt="image" src="https://user-images.githubusercontent.com/5634097/227053101-f4971dcd-9fee-481f-a991-94a0f8ada17a.png">


@amanagr another question -- now that the messages have background color different than the main background, the corners of them peek out of recipient bar

<img width="887" alt="image" src="https://user-images.githubusercontent.com/5634097/227052724-ec9858b9-f192-4d44-a447-b11814a6159c.png">

Do you have ideas about how to fix that, given the way it's currently structured? Is there a nice way to have the sticky header get more vertical space than the messages can?

amanagr commented:
I updated #23782 with a fix.

evykassirer commented:
Awesome! Looks like this now

<img width="1070" alt="image" src="https://user-images.githubusercontent.com/5634097/227364550-e7a7bca8-3451-4fb4-9b93-98040a39502d.png">

Next step is either:

* @alya or @terpimost re-checking this version since the integration into Aman's changes
* @timabbott just pulling it into a test deploy

alya commented:
Thanks!

1. I'm pretty sure the top bar background is supposed to extend all the way across?
2. The background for date dividers in PMs is not correct:

![Screen Shot 2023-03-23 at 3 03 04 PM](https://user-images.githubusercontent.com/2090066/227375867-b652d7dd-4d24-4da1-a370-b6d1541fc5cf.png)

alya commented:
Eep!

![Screen Shot 2023-03-23 at 3 11 15 PM](https://user-images.githubusercontent.com/2090066/227376798-f9e3cc00-cd4e-46c6-8034-35745240bbd6.png)

alya commented:
The borders in Drafts are off, even more so than [on CZO](https://chat.zulip.org/#narrow/stream/101-design/topic/UI.20redesign.3A.20stream.2Ftopic.20header.20bar/near/1531980) right now.

![Screen Shot 2023-03-23 at 3 11 52 PM](https://user-images.githubusercontent.com/2090066/227376916-1c3498f8-5e35-4fcd-b272-e29eb18ec265.png)

alya commented:
That's all I've got in this round.

evykassirer commented:
1. I think the top bar doesn't extend because of the scrollbar, but maybe @terpimost can confirm
 
<img width="1015" alt="image" src="https://user-images.githubusercontent.com/5634097/227377085-a5b6c5fc-8986-47d3-8ca3-14bf7f4932ad.png">

2. Thanks! Interesting that this showed up in this PR instead of the message background PR (where it's not an issue). Ah, it's because the PM colors already changed and I forgot to account for that in this PR, heh. Fixing it now :)

<img width="976" alt="image" src="https://user-images.githubusercontent.com/5634097/227377611-d2037711-eaee-4c6a-ac17-a840749387d3.png">

3. This is approximately the same as on CZO, just for both stream and PM instead of just PM because now both message-types have background-color (similar to a few other bugs Aman has been working with me on).

<img width="770" alt="image" src="https://user-images.githubusercontent.com/5634097/227378262-d07b4fc2-e610-4777-a461-1b044cc0ab1d.png">

Whatever Aman does to fix it will fix it here too!

---

I'll look into the modal stuff now.

evykassirer commented on package.json:
I need to remove this and the pnpm-lock files.

evykassirer commented:
overlay screenshots

<details>

<img width="760" alt="image" src="https://user-images.githubusercontent.com/5634097/227379726-bc333664-e5ea-4f1b-86b5-aae8d869ebe8.png">

<img width="708" alt="image" src="https://user-images.githubusercontent.com/5634097/227379763-16e45a27-91fb-401f-ad29-f75def2dd0f9.png">

<img width="1096" alt="image" src="https://user-images.githubusercontent.com/5634097/227379808-759b0754-98f5-4371-aaf8-40d9a238b6fe.png">

</details>


stream messages

<details>

<img width="1266" alt="image" src="https://user-images.githubusercontent.com/5634097/227379910-b206b8b9-9b3a-4a02-ac79-859ffd70e351.png">

<img width="1269" alt="image" src="https://user-images.githubusercontent.com/5634097/227380287-148e18e7-fa7e-4a6a-9199-7729a2141e44.png">

<img width="1265" alt="image" src="https://user-images.githubusercontent.com/5634097/227382061-e6891633-933b-4119-8e84-047786698122.png">


</details>

private messages

<details>

<img width="1257" alt="image" src="https://user-images.githubusercontent.com/5634097/227380355-ef58dd95-11a9-42d2-86f3-21400994b9f2.png">

<img width="1265" alt="image" src="https://user-images.githubusercontent.com/5634097/227380394-9feee20d-2291-4586-b81a-d240b22319a6.png">

<img width="1071" alt="image" src="https://user-images.githubusercontent.com/5634097/227382904-6c69416c-143f-4e0a-8451-3a1799f98388.png">


</details>

recent conversations -- it's a little ugly, but I'm not sure if it's worth changing things much before we redesign it. might be worth asking Vlad about it

<details>

<img width="1256" alt="image" src="https://user-images.githubusercontent.com/5634097/227380471-67ed998e-0750-4e73-a14d-08ba0e2b2b20.png">

<img width="1248" alt="image" src="https://user-images.githubusercontent.com/5634097/227383060-48795e6b-930d-47f9-ac4b-0040436112c7.png">

<img width="636" alt="image" src="https://user-images.githubusercontent.com/5634097/227383154-7de0ebb6-9674-4f5b-8e62-d4b0fd41b712.png">

</details>

friends list open looks weird but same as CZO

<details>

<img width="391" alt="image" src="https://user-images.githubusercontent.com/5634097/227381554-dd5a9659-829e-42f2-8359-08559344c21a.png">

<img width="422" alt="image" src="https://user-images.githubusercontent.com/5634097/227381540-fbb9be58-849d-40d5-8219-6321631d59a8.png">

<img width="367" alt="image" src="https://user-images.githubusercontent.com/5634097/227383212-8f47ec18-6f36-440e-a284-43ea5630b91a.png">


</details>

stream settings -- the stream color-changer is a little strange, but it's also quite strange on CZO in both light and dark mode

<details>

<img width="628" alt="image" src="https://user-images.githubusercontent.com/5634097/227381311-7f43cf6b-79a2-4b2c-a2c8-b3de2da9f7df.png">

<img width="1232" alt="image" src="https://user-images.githubusercontent.com/5634097/227383296-87bb23f9-aea3-460b-b709-78269f4b23b0.png">

</details>

compose box

<details>

<img width="967" alt="image" src="https://user-images.githubusercontent.com/5634097/227383400-fb00d8db-20a1-4fda-88b2-f0c674b45057.png">

<img width="969" alt="image" src="https://user-images.githubusercontent.com/5634097/227383438-da06d093-2c2e-4e4b-93bb-2eec088df3cc.png">

</details>

Drafts aren't affected except the message background thing that's related to the bug Aman will fix.

evykassirer commented:
@alya let me know if there are main areas of the app you'd test that aren't mentioned in that list and I can add them before you look again!

evykassirer commented:
oh boy I have some git issue I need to undo 😅 but the visual testing should be the same

alya commented:
> friends list open looks weird but same as CZO

Hm, it doesn't look like that for me on CZO (but I haven't checked on the PR, so maybe that will look OK too):

![Screen Shot 2023-03-23 at 4 30 14 PM](https://user-images.githubusercontent.com/2090066/227386853-deaaab47-8406-419f-baed-b170b8941dae.png)

alya commented:
Thanks! I would definitely add wide-screen screenshots, since all those look pretty narrow. Probably with both configurations of "Use full width on wide screens".

alya commented:
I also did more poking at the compose box, including trying it with PMs vs stream messages, in full-height mode, and with some banners.

alya commented:
You may also want to check the "more topics" view in the left sidebar.

alya commented:
I also tested @-mentions of the user (since those have a colored background) .

alya commented:
This is all without looking at the code -- you of course know better than I do about what's likely to be affected by any specific PR.

evykassirer commented:
Thanks, I'll add those! 

The code unfortunately doesn't help me much for this PR, since the bugs are likely things that I _didn't_ change but should have, since the code is pretty much just changing the colors of background, modal backgrounds, messages, and the compose box.

evykassirer commented:
**wide screens**

modal/overlay -- no change from narrow

stream messages

<details>

<img width="1689" alt="image" src="https://user-images.githubusercontent.com/5634097/227394167-a4e575cc-0698-4d85-aec3-0ed0a38c3977.png">

<img width="1769" alt="image" src="https://user-images.githubusercontent.com/5634097/227394199-1f3324ee-8a16-4644-91f0-011dd9a49ba9.png">



</details>

private messages

<details>

<img width="1677" alt="image" src="https://user-images.githubusercontent.com/5634097/227394263-7354e2a2-6309-408c-b860-d358646c0df0.png">



</details>

recent conversations

**asked Vlad about this**

<details>

<img width="1843" alt="image" src="https://user-images.githubusercontent.com/5634097/227394745-52a3179a-760d-452d-b942-062f8536a73a.png">

<img width="1830" alt="image" src="https://user-images.githubusercontent.com/5634097/227396833-8499a2e8-4abf-4b76-a242-677c8fca629d.png">


</details>

settings menu

<details>

<img width="627" alt="image" src="https://user-images.githubusercontent.com/5634097/227395041-b3a90493-5bc5-49ed-aee9-313b04825c65.png">

<img width="423" alt="image" src="https://user-images.githubusercontent.com/5634097/227396584-8c098e9b-b6a2-43fe-be44-1c6c2b4aa5e1.png">

This might not be the final color we choose, but it looks similar to the designs and there will be search redesign work done.

</details>

compose box

<details>

<img width="1082" alt="image" src="https://user-images.githubusercontent.com/5634097/227395188-f825c1b2-3f1b-49ca-9697-2dcee1248a4d.png">

<img width="1098" alt="image" src="https://user-images.githubusercontent.com/5634097/227396538-9c72d9b9-66a7-4476-9e3e-5ebb00e55e73.png">


</details>

----

more compose box states

<details>

<img width="1127" alt="image" src="https://user-images.githubusercontent.com/5634097/227395234-d23c7835-3154-4ea7-a3eb-5916416e2e79.png">

<img width="1151" alt="image" src="https://user-images.githubusercontent.com/5634097/227396378-f969ede6-5559-4eac-b2bc-b3898857c519.png">


<img width="1114" alt="image" src="https://user-images.githubusercontent.com/5634097/227395297-0d11fdfe-ec1c-4eef-ad28-68ea081ea130.png">

<img width="1119" alt="image" src="https://user-images.githubusercontent.com/5634097/227396414-b50f6855-b292-48c5-95c6-46458feb4b92.png">


<img width="1129" alt="image" src="https://user-images.githubusercontent.com/5634097/227395550-f74a37b8-3633-43c1-846a-70af9abefa48.png">

<img width="1096" alt="image" src="https://user-images.githubusercontent.com/5634097/227396443-0baa7367-6c5a-4a1d-9a2c-ac358bc64416.png">

<img width="1097" alt="image" src="https://user-images.githubusercontent.com/5634097/227396468-43fa2fa1-d5a6-4671-a5c3-00d91bdd5b04.png">

(I'm reminded we still need to move the banner out of the compose box)

</details>

"more topics" view in the left sidebar

<details>

<img width="350" alt="image" src="https://user-images.githubusercontent.com/5634097/227395852-3b9fd1a8-8d3a-4b9f-ac1f-519b784d63dc.png">

found and fixed a bug in dark mode!

<img width="461" alt="image" src="https://user-images.githubusercontent.com/5634097/227396289-5a4d5983-b67f-4af1-8852-c5f337e72f84.png">

(dark mode on CZO also has no border, so that's fine)

</details>

@-mentions of the user

<details>

<img width="1081" alt="image" src="https://user-images.githubusercontent.com/5634097/227395901-9264b727-7d37-444f-88c8-1e427b97e252.png">

<img width="1145" alt="image" src="https://user-images.githubusercontent.com/5634097/227396046-fcd6f5de-7f6f-4b9a-948b-bb3138bf7fa2.png">


</details>

empty view -- not totally sure what you meant, is it this?

<details>

<img width="1798" alt="image" src="https://user-images.githubusercontent.com/5634097/227395954-b009acdb-0a9c-4a42-9334-0757c551283a.png">

<img width="1800" alt="image" src="https://user-images.githubusercontent.com/5634097/227396008-7c4fe4f6-bf89-4d1c-880f-4b3fa2c52db4.png">


</details>

still to do:

* full width header on really big screens

alya commented:
> empty view -- not totally sure what you meant, is it this?

Yep, that's what I had in mind.

alya commented:
Thanks for going through all those!

evykassirer commented:
Here's full width header! I also left-justified the zulip logo because it looked weird with it centered (and this changes moves us toward the design we're moving towards)

<img width="1002" alt="image" src="https://user-images.githubusercontent.com/5634097/227742615-a2a30517-8db2-4be7-a514-d1ab22aeda01.png">

<img width="1920" alt="image" src="https://user-images.githubusercontent.com/5634097/227742621-bf23c942-615a-45eb-9751-94ecbc5f2663.png">

evykassirer commented:
For Alya: I've addressed all issues I've found so far, and posted screenshots for the full-width vabar

On the techincal side:

* Once Aman rebases his PR on the color changes, I'll rebase mine as well (I don't want to do the work to pull the changes through is commits)
* The change I made for the full-width navbar (most recent commit) could possibly be done without adding a new div. @andersk do you have any ideas?

alya commented:
Thanks,, the screenshots look good to me. I'll plan to do manual testing after the rebase.

terpimost commented:
it would be nice to have darker border 1 line under the header It could be done with shadow, like this:
```
box-shadow: inset 0px -1px 0px #CCCCCC;
```
for dark version:
```
box-shadow: inset 0px -1px 0px rgba(0, 0, 0, 0.6);
```

evykassirer commented:
@terpimost can you show me a screenshot / Figma design of what you mean?

evykassirer commented:
updates:

* color changes are pulled in, no conflicts
* Alya could take a look now, or could wait for the conversation about the darker/thickened border to finish

alya commented:
> Alya could take a look now, or could wait for the conversation about the darker/thickened border to finish

Hm, I guess I'll wait until tomorrow, in case there are updates.

evykassirer commented:
weird things going on I need to investigate, will update when it's ready again.

evykassirer commented:
It's the in main commit, so I'm not sure why it says that

timabbott commented:
Is this heading section of "Recent conversations" that set of colors by design? It looks leftover from the previous version.

![image](https://user-images.githubusercontent.com/2746074/231615639-06b586cb-2fa4-460d-bd9d-43fa6e8b46c5.png)

![image](https://user-images.githubusercontent.com/2746074/231615672-cb423191-ed4c-4ea2-ad4d-eaede1259cea.png)

timabbott commented on web/styles/zulip.css:
```diff
@@ -295,11 +299,14 @@ p.n-margin {
+.header-main,
+#message_view_header {
+    background-color: hsl(0deg 0% 96%);
 }
```
This set of items seems a bit random. It might be helpful to both define a variable for this color (maybe `--navbar-background`?) at least; doing so might also save some work in the dark theme CSS file.

evykassirer replied:
Ah yeah I worked on this before we started making more constants for colors -- I can add them here.

timabbott commented on web/styles/zulip.css:
```diff
@@ -1092,6 +1099,7 @@ td.pointer {
     vertical-align: top;
     border: none;
     padding: 0 5px;
+    background-color: hsl(0deg 0% 100%);
```
I see this color repeated a bunch; should it have a name too?

timabbott commented on web/styles/dark_theme.css:
```diff
@@ -720,8 +734,11 @@
-    li.active-sub-filter {
+    & li.active-filter {
+        background-color: var(--color-background);
+    }
```
I'm not sure if this line is what's responsible, but the behavior of hovering over the current stream narrow in the left sidebar is weird in a way that looks like a selector priority bug -- in the dark theme there's no change in appearance on hover at all.

evykassirer replied:
thanks for pointing this out -- fixed now!


https://user-images.githubusercontent.com/5634097/231871539-bd809934-896b-4f05-ba79-16bdc6a011e0.mp4


https://user-images.githubusercontent.com/5634097/231871567-8e83abe2-4afd-4499-8447-ca1663575e9d.mp4

timabbott commented on web/templates/navbar.hbs:
```diff
@@ -1,73 +1,76 @@
-                </div>
-                <div class="top-navbar-border top-navbar-container">
-                    <div id="message_view_header" class="notdisplayed">
+    {{!-- remove `top_navbar_full_width` wrapper div once the scrollbar is on the `html` element, and then move the styles on `header` to `navbar-container --}}
```
Which scrollbar are we expecting to move to the `html` element? I'm not sure what future plan this is referring to.

timabbott replied:
And do we really need an extra div rather than applying styles to either the `header` or `header-main` element? I am fine with it if we do, but I also don't understand what this div does, since it appears to have no styles that would make it a different size from the elements around it?

evykassirer replied:
Maybe you can help me write a clearer comment, but here's a video of what I mean. The scrollbar is currently on `app`, so the navbar needs to have height 0. `header` has bottom padding that needs to be the app background color, and `top_navbar` is centered on big screens with auto margin on its sides so it can't fill the full width with the header color.

https://user-images.githubusercontent.com/5634097/231659007-890d5f39-5a55-4f9e-81b6-dbca9a9f9d4f.mp4




red is `header`. I added a wrapper around `top_navbar` to make its color full width

<img width="782" alt="image" src="https://user-images.githubusercontent.com/5634097/231658912-508a70aa-6f9f-476c-b3f4-3c244b3de037.png">

evykassirer commented:
I asked about Recent Conversations on CZO [here](https://chat.zulip.org/#narrow/stream/101-design/topic/background.20color/near/1532160) and didn't get a response and just followed up again now.

evykassirer commented:
I'm planning to address the three actionable comments tomorrow.

evykassirer commented:
updated recent conversations view in dark mode:

![image](https://user-images.githubusercontent.com/5634097/231872499-26bb1660-94db-4630-9833-6817c97fb862.png)

timabbott commented:
Test-deployed this on chat.zulip.org.

evykassirer commented:
changes discussed [here](https://chat.zulip.org/#narrow/stream/101-design/topic/UI.20redesign.3A.20background/near/1549838)

alya commented:
Thanks for the updates! I don't think the "more topics" fix is working correctly; it should be possible to get the blue background on top of "Streams" like this:

<img width="291" alt="Screen Shot 2023-04-17 at 10 05 00 PM" src="https://user-images.githubusercontent.com/2090066/232676335-a1f4ba2b-53d9-46e8-9cf8-f9f7ad50e502.png">

evykassirer commented:
@alya I'm having trouble reproducing that in a very short window.

alya commented:
Try scrolling the left sidebar down in a short window. It replicates for me...

alya commented:
(It probably doesn't need to be short -- just needs to have enough topics to scroll.)

evykassirer commented:
![Kapture 2023-04-22 at 17 19 39](https://user-images.githubusercontent.com/5634097/233812860-7120f135-2d4a-4688-a649-d7c5b912d256.gif)

@alya are you doing something I'm not doing?

alya commented:
Hm, I don't know if the latest set of changes have fixed this issue -- it's currently not reproducing for me.

alya commented:
Testing the current version of the PR, there seems to be an extraneous stripe on the right in the narrow version of the Recent conversations view:

![Screen Shot 2023-04-24 at 8 51 48 AM](https://user-images.githubusercontent.com/2090066/234050227-69820619-03fd-4fd9-98cf-a19bb8f98eb2.png)

alya commented:
Actually, in other views too... I guess it's where the scrollbar goes, but it's there even when there is no scroll bar, which seem odd?

evykassirer commented:
I believe that's always been there including before this change.

evykassirer commented:
Most recent push includes a commit that addresses all but the last two parts of #25247 (the rest are covered by Aman in another PR)

timabbott commented:
Can you rebase this? Seems to merge conflict a bit with https://github.com/zulip/zulip/pull/25262 that I just merged, and also with the unread marker color change (#25137), which based on your changes I'm not sure whether is incorporated into this in some form or not?

evykassirer commented:
Rebased off of main and split the commit of Vlad's updates into the two that people were happy with and the one that people weren't as happy with (sidebar active filter), for easier deploy.

timabbott commented:
OK updating the deploying of this skipping that last commit, since I think that was what you were suggesting.

timabbott commented:
Updating the deployment of this.

timabbott commented:
Updating the chat.zulip.org deployment for that latest bug fix.

timabbott commented:
> temp: Vlad's suggestions for "Hover and active styles in a side bar" 

This is the commit I'm supposed to keep not deploying because it's not for 7.0, right? If so, let's plan to drop this commit from the PR.

evykassirer commented:
I think we can keep it? I just squashed it in. Or do you deliberately not want to change that?

(4323b60a7da318a3a412f53498ae0751248ad417 is the commit for reference, if we want to pull it back out)

evykassirer commented:
Actually to keep things simple, I switched the colors back to how they were before. The color variables are still there if we want to change the values back.

evykassirer commented:
Fixed conflicts

timabbott commented:
Updating chat.zulip.org with this.

timabbott commented:
Merged, thanks @evykassirer and @amanagr for all the work on getting this into an integrable state!

I suspect that this change (along with the recipient bar one merged earlier) might be the sloggiest part of the redesign project (in terms of how many types of weird problems it can cause), so I'm glad to have it behind us.

andersk commented on templates/zerver/app/index.html:
```diff
@@ -51,7 +51,7 @@
     }
     :root.dark-theme #app-loading {
-    background-color: hsl(212, 28%, 18%);
+    background-color: hsl(0, 0%, 11%);
```
This needs to be duplicated into the corresponding `.color-scheme-automatic` block for the rebase on #25504.
